card/src/main/java/com/adyen/checkout/card/AddressDelegate.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ozgur on 18/3/2022.
+ */
+
+package com.adyen.checkout.card
+
+import android.util.LruCache
+import com.adyen.checkout.card.api.AddressDataType
+import com.adyen.checkout.card.api.makeUrl
+import com.adyen.checkout.card.api.model.AddressItem
+import com.adyen.checkout.card.repository.AddressRepository
+import com.adyen.checkout.card.ui.AddressFormInput
+import com.adyen.checkout.components.base.Configuration
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+
+class AddressDelegate(
+    private val addressRepository: AddressRepository
+) {
+
+    private val _statesFlow: MutableStateFlow<List<AddressItem>> = MutableStateFlow(emptyList())
+    internal val statesFlow: Flow<List<AddressItem>> = _statesFlow
+
+    private val cache: LruCache<String, List<AddressItem>> = LruCache<String, List<AddressItem>>(CACHE_ENTRY_SIZE)
+
+    fun getStateList(
+        configuration: Configuration,
+        countryCode: String?,
+        coroutineScope: CoroutineScope
+    ) {
+        val addressSpecification = AddressFormInput.AddressSpecification.fromString(countryCode)
+        val needsStates = addressSpecification == AddressFormInput.AddressSpecification.BR ||
```

Comment:
Maybe we could make an array that contains BR, US and CA and change `CACHE_ENTRY_SIZE` to use this array's length + 1